README.md

```diff
@@ -51,6 +51,132 @@ After this, you will have a directory containing files used for creating a
 custom Jupyter widget. You will now be able to easily package and distribute
 your custom Jupyter widget.
 
+## Files Generated by cookiecutter
+Jupyter widgets are made of two parts:
+- Python package for the code executed in kernel as the back-end of the widget
+- Javascript NPM package for the front-end of the widget running in browser
+
+cookiecutter generates source files needed for both back-end and front-end of the widget
+
+```
+├── <python_package_name>
+│   ├── __init__.py
+│   ├── _version.py
+│   └── example.py
+├── js
+│   ├── lib
+│   │   ├── embed.js
+│   │   ├── example.js
+│   │   ├── extension.js
+│   │   ├── index.js
+│   │   └── labplugin.js
+│   ├── package.json
+│   ├── README.md
+│   └── webpack.config.js
+├── .gitignore
+├── <npm_package_name>.json
+├── MANIFEST.in
+├── README.md
+├── RELEASE.md
+├── setup.cfg
+└── setup.py
+```
+
+- **`<python_package_name> directory`**\
+This directory contains the python package that runs in the kernel
+
+    - **`__init__.py`**\
+Marks the `<python_package_name>` directory as a python module, allowing `<python_package_name>` to be imported as a module
+
+    - **`_version.py`**\
+Contains python module version definitions
+
+    - **`example.py`**\
+Widget class definition in python. Defines widget data model that is synced with front-end automatically using traitlets and Backbone.js
```

Comment:
Maybe just:

Definition of Python interface to the widget. This file defines the Python interface, and which parts of the data are synced to the frontend.

(I don't think there's value in mentioning just traitlets and Backbone.js, since there's tons of components that take part in the synchronization)